### ISSUE TITLE
Parts connections get mixed up

### ISSUE THREAD

**Issue by DrprofLuigi** [DonBruce64/MinecraftTransportSimulator#1328]
There seems to be an issue with parts connecting that shouldn't be. Trailers seem to get their connections messed up. When I have two trailers connected, and I click to release one, both will disconnect. Same with props, when I click one, both spin (on both BETA 8 and 9). I can even start one engine from the other.

Here is an example json from the updated UNU Navigator (the first engine is part 4):

```		{
			"pos" : [2.875, 0, 3.59375],
			"types" : ["engine_radial"],
			"defaultPart": "unuparts:unuparts_part_unu_engine_r09",
			"minValue" : 0.75,
			"maxValue" : 1.2,
			"intakeOffset": -0.5,
			"interactableVariables": [["!part_present_5"]]
		},
				{
					"pos" : [2.875, 0, 4.125],
					"types" : ["propeller"],
					"defaultPart": "unuparts:unuparts_part_unu_propeller_3d054",
					"minValue" : 133,
					"maxValue" : 133,
					"interactableVariables": [["part_present_4"]],
					"linkedParts" : [4],
					"animations": [
						{
							"animationType": "rotation",
							"centerPoint": [2.875, 0, 4.125],
							"axis": [0, 0, -1],
							"variable": "propeller_rotation_1"
						},
						{
							"animationType": "rotation",
							"centerPoint": [2.875, 0, 4.125],
							"axis": [0, 0, 1],
							"variable": "engine_rotation_1"
						}
					]
				},
		{
			"pos" : [-2.875, 0, 3.59375],
			"types" : ["engine_radial"],
			"defaultPart": "unuparts:unuparts_part_unu_engine_r09",
			"minValue" : 0.75,
			"maxValue" : 1.2,
			"intakeOffset": -0.5,
			"interactableVariables": [["!part_present_7"]]
		},
				{
					"pos" : [-2.875, 0, 4.125],
					"types" : ["propeller"],
					"defaultPart": "unuparts:unuparts_part_unu_propeller_3d054",
					"minValue" : 133,
					"maxValue" : 133,
					"interactableVariables": [["part_present_6"]],
					"linkedParts" : [6],
					"animations": [
						{
							"animationType": "rotation",
							"centerPoint": [-2.875, 0, 4.125],
							"axis": [0, 0, -1],
							"variable": "propeller_rotation_2"
						},
						{
							"animationType": "rotation",
							"centerPoint": [-2.875, 0, 4.125],
							"axis": [0, 0, 1],
							"variable": "engine_rotation_2"
						}
					]
				},```

**Comment by DrprofLuigi** [comment]
There is a very similar issue with the Aurornis. All props spin regardless of which engine is on. These props also appear to produce thrust.

**Comment by DonBruce64** [comment]
Fixed in V22.2.0.
